apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceContext.java

```diff
@@ -293,6 +296,14 @@ void setRecorded(boolean recorded) {
         }
     }
 
+    public void setDiscard(boolean discard) {
+        this.discard = discard;
```

Comment:
We have a similar flag in `Transaction` called `noop`. Maybe we should rather pull that up to `AbstractSpan`?

Reply:
Seems like currently they effectively serve a similar purpose. I will move it and if we see a new need to differentiate `noop` from `discard` we can reiterate later.

Reply:
Actually, this breaks the logic used for this flag- `noop` transactions are not-sampled, and this is an inheritable trait, so all decedents will be non-sampled. However, marking a span as `discard` should not be inherited and should not affect parent (as opposed to the `discard` flag).